core/Fetch.cpp

```diff
@@ -105,16 +106,28 @@ namespace olympia
         fetch_inst_event_->schedule(sparta::Clock::Cycle(0));
     }
 
-    // Called from Retire via in_fetch_flush_redirect_ port
-    void Fetch::flushFetch_(const uint64_t & new_addr) {
-        ILOG("Fetch: receive flush on new_addr=0x"
-             << std::hex << new_addr << std::dec);
+    // Called from FlushManager via in_fetch_flush_redirect_port
+    void Fetch::flushFetch_(const FlushManager::FlushingCriteria &criteria)
+    {
+        ILOG("Fetch: received flush " << criteria);
+
+        auto flush_inst = criteria.getInstPtr();
+
+        // Rewind the tracefile
```

Comment:
You can remove the if-statement here by passing `!criteria.isInclusiveFlush()` as the second parameter to `reset` instead.